src/spiffe_id/mod.rs

```diff
@@ -90,29 +85,47 @@ impl SpiffeId {
     /// ```
     /// use spiffe::spiffe_id::SpiffeId;
     ///
-    /// let spiffe_id = SpiffeId::new("spiffe://example.org/path").unwrap();
-    /// assert_eq!("example.org", spiffe_id.trust_domain().to_string());
+    /// let spiffe_id = SpiffeId::new("spiffe://trustdomain/path").unwrap();
+    /// assert_eq!("trustdomain", spiffe_id.trust_domain().to_string());
     /// assert_eq!("/path", spiffe_id.path());
     /// ```
     pub fn new(id: &str) -> Result<Self, SpiffeIdError> {
-        let id = id.trim();
         if id.is_empty() {
             return Err(SpiffeIdError::Empty);
         }
 
-        let url = Url::from_str(id)?;
-        Self::validate_spiffe_id(&url)?;
+        if !id.contains(SCHEME_PREFIX) {
```

Comment:
Maybe:
```rust
if id[..SCHEME_PREFIX.len()] != SCHEME_PREFIX {
    ...
}
```
as `contains` could check in any substring? Not that it might be very important...